ddpui/core/ai/tools/profile_tools.py

```
@@ -0,0 +1,41 @@
+        catalog.check_table(ctx, schema_name, table_name)
+    except catalog.ToolInputError as err:
+        return str(err)
+
+    if not ctx.warehouse.column_exists(schema_name, table_name, column_name):
+        return f"Column '{column_name}' does not exist on {schema_name}.{table_name}. Use get_table_details to see columns."
+
+    qualified = catalog.qualified(ctx.dialect, schema_name, table_name)
+    quoted_col = f"`{column_name}`" if ctx.dialect == "bigquery" else f'"{column_name}"'
+    sql = (
+        f"SELECT {quoted_col} AS value, COUNT(*) AS occurrences FROM {qualified} "
+        f"GROUP BY 1 ORDER BY 2 DESC LIMIT {TOP_VALUES_COUNT}"
```

Ruff (0.15.21)

[error] Possible SQL injection vector through string-based query construction

(S608)